src/app/resperson/get.controller.js

```diff
@@ -0,0 +1,38 @@
+const logger = require('../../common/utils/logger')(__filename);
+const CookieModel = require('../../common/models/Cookie.class');
+const resPersonApi = require('../../common/services/resPersonApi');
+
+module.exports = async (req, res) => {
+  logger.debug('In responsible person get controller');
+  const cookie = new CookieModel(req);
+  const errMSg = { message: 'Failed to get saved responsible persons' };
+  try {
+    const response = await resPersonApi.getResPersons(cookie.getUserDbId())
+    const resPersons = JSON.parse(response);
+    if (req.session.successMsg) {
+      const { successMsg, successHeader } = req.session;
+      delete req.session.successHeader;
+      delete req.session.successMsg;
+      return res.render('app/resperson/index', { cookie, resPersons, successMsg, successHeader });
+    }
+    if (req.session.errMsg) {
+      const { errMsg } = req.session;
+      delete req.session.errMsg;
+      return res.render('app/resperson/index', {
+        cookie, resPersons, errors: [errMsg],
+      });
+    }
+    if(resPersons.message) {
+      logger.info(`Failed to get saved people: ${resPersons.message}`)
```

Comment:
This `if(resPersons.message) {` is odd:
* As far as I am away `resPersonApi.getResPersons` --> `/<user_id>/responsiblepersons` does nto return delibrate erors inthe `{ "message": xyz }` format.
* If it did, it would cause a minor bug with `src/app/resperson/index.njk` as `resPersons` is an object, objects can have `|` [length](https://mozilla.github.io/nunjucks/templating.html#length).
* The log message is incorrect as well with `Failed to get saved people:`
```
    {% if resPersons | length %}
              {% for resPerson in resPersons %}
 ```

Reply:
1. It do returns an error message { "message": xyz } format, it is coming from backend, it is not very obvious but it is coming from @validate_user type.
2. resPersons is not an object here it is an array/list, list of responsible persons for a user
3.  will update the log message.

Reply:
Point 1 and 3 and are good.

On point 2, I am uncertain. I though that if `resPersons.message` was `true`, then `resPersons` is necessarily an object, otherwise it would not have the `resPersons.message` e.g. below:
```js
    const response = await resPersonApi.getResPersons(cookie.getUserDbId())
    const resPersons = JSON.parse(response); // {'message': 'User not registered'}
    if (req.session.successMsg) {
      const { successMsg, successHeader } = req.session;
      delete req.session.successHeader;
      delete req.session.successMsg;
      return res.render('app/resperson/index', { cookie, resPersons, successMsg, successHeader });
    }
    if (req.session.errMsg) {
      const { errMsg } = req.session;
      delete req.session.errMsg;
      return res.render('app/resperson/index', {
        cookie, resPersons, errors: [errMsg],
      });
    }
    if(resPersons.message) {
      logger.info(`Failed to get saved responsible persons: ${resPersons.message}`)
      return res.render('app/resperson/index', {
        cookie, 
        resPersons, // this would be {'message': 'User not registered'}
        errors: [errMessage],
      });
    }
    return res.render('app/resperson/index', { cookie, resPersons });
    ```

Reply:
That is fine.